samples/rbac.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cosi-controller-sa
+  # replace with non-default namespace name
+  namespace: default
+
+---
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cosi-controller-manager-role
+rules:
+  - apiGroups: ["objectstorage.k8s.io"]
+    resources: ["bucketrequests", "bucketaccessrequests"]
+    verbs: ["get", "list", "watch", "create", "delete"]
+  - apiGroups: ["objectstorage.k8s.io"]
+    resources: ["buckets", "bucketaccess"]
+    verbs: ["get", "list", "watch", "update", "create", "delete"]
+  - apiGroups: ["objectstorage.k8s.io"]
+    resources: ["buckets","bucketrequests", "bucketclass","bucketaccess","bucketaccessrequests", "bucketaccessclass"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cosi:system
+subjects:
+  - kind: ServiceAccount
+    name: default
+    namespace: default
+roleRef:
+  kind: ClusterRole
+  name: cosi-controller-manager-role
+  apiGroup: rbac.authorization.k8s.io
+
+---
+
+# Controller must be able to work with endpoints in current namespace
+# if (and only if) leadership election is enabled
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  # replace with non-default namespace name
+  namespace: default
+  name: cosi-controller-manager
+rules:
+# Only one of the following rules for endpoints or leases is required based on
+# what is set for `--leader-election-type`. Endpoints are deprecated in favor of Leases.
+- apiGroups: [""]
+  resources: ["endpoints"]
```

Comment:
Probably don't understand the comment above, but if endpoints are deprecated in favor of leases, and you include leases below, why do we need enpoints here?

Reply:
Fixed